homeassistant/components/awattar/sensor.py

```diff
@@ -0,0 +1,196 @@
+        suggested_display_precision=4,
+        native_unit_of_measurement=f"{CURRENCY_EURO}/{UnitOfEnergy.KILO_WATT_HOUR}",
+        value_fn=lambda data: data.awattar.for_timestamp(dt_util.now()).price_per_kWh,
+    ),
+    AwattarSensorEntityDescription(
+        key="next_hour_price",
+        translation_key="next_hour_price",
+        state_class=SensorStateClass.MEASUREMENT,
+        suggested_display_precision=4,
+        native_unit_of_measurement=f"{CURRENCY_EURO}/{UnitOfEnergy.KILO_WATT_HOUR}",
+        value_fn=lambda data: (
+            data.awattar.for_timestamp(dt_util.now() + timedelta(0, 3600)).price_per_kWh
+        ),
+    ),
+    AwattarSensorEntityDescription(
+        key="average_price",
+        translation_key="average_price",
+        state_class=SensorStateClass.MEASUREMENT,
+        suggested_display_precision=4,
+        native_unit_of_measurement=f"{CURRENCY_EURO}/{UnitOfEnergy.KILO_WATT_HOUR}",
+        value_fn=lambda data: data.awattar.mean().price_per_kWh,
+    ),
+    AwattarSensorEntityDescription(
+        key="max_price",
+        translation_key="max_price",
+        state_class=SensorStateClass.MEASUREMENT,
+        suggested_display_precision=4,
+        native_unit_of_measurement=f"{CURRENCY_EURO}/{UnitOfEnergy.KILO_WATT_HOUR}",
+        value_fn=lambda data: data.awattar.max().price_per_kWh,
+    ),
+    AwattarSensorEntityDescription(
+        key="min_price",
+        translation_key="min_price",
+        state_class=SensorStateClass.MEASUREMENT,
+        suggested_display_precision=4,
+        native_unit_of_measurement=f"{CURRENCY_EURO}/{UnitOfEnergy.KILO_WATT_HOUR}",
+        value_fn=lambda data: data.awattar.min().price_per_kWh,
+    ),
+    AwattarSensorEntityDescription(
+        key="slot_2_hrs_start",
+        translation_key="slot_2_hrs_start",
+        device_class=SensorDeviceClass.TIMESTAMP,
+        value_fn=lambda data: data.awattar.best_slot(2).start_datetime,
+    ),
+    AwattarSensorEntityDescription(
+        key="slot_3_hrs_start",
+        translation_key="slot_3_hrs_start",
+        device_class=SensorDeviceClass.TIMESTAMP,
+        value_fn=lambda data: data.awattar.best_slot(3).start_datetime,
+    ),
+    AwattarSensorEntityDescription(
+        key="slot_4_hrs_start",
+        translation_key="slot_4_hrs_start",
+        device_class=SensorDeviceClass.TIMESTAMP,
+        value_fn=lambda data: data.awattar.best_slot(4).start_datetime,
+    ),
+    AwattarSensorEntityDescription(
+        key="slot_5_hrs_start",
+        translation_key="slot_5_hrs_start",
+        device_class=SensorDeviceClass.TIMESTAMP,
+        value_fn=lambda data: data.awattar.best_slot(5).start_datetime,
+    ),
+    AwattarSensorEntityDescription(
+        key="slot_6_hrs_start",
+        translation_key="slot_6_hrs_start",
+        device_class=SensorDeviceClass.TIMESTAMP,
+        value_fn=lambda data: data.awattar.best_slot(6).start_datetime,
+    ),
```

Comment:
Do we want all those sensors, or would this data be more suitable for a service which returns a the requested time slot and the price?

Reply:
Hi there & thanks for your comments. If we change it, in my opinion, all integrations that provide this kind of information should do it the same way. Next question:  if it is a service, how could i display this information in a dashboard and/or use it in automations (e.g. i want to start charging something when the cheapest 3hour period starts...)